packages/components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,221 @@
+                            />
+                        ) : null}
+                        <IconButton
+                            name={isOptionsVisible ? "menu-up" : "menu-down"}
+                            variant="ghost"
+                            iconSize={18}
+                            onPress={handleMenuOpen}
+                            style={styles.menuIcon}
+                        />
+                    </View>
+                }
+            />
+            {isOptionsVisible && (
+                <Menu
+                    anchorEl={inputRef}
+                    open={isOptionsVisible}
+                    onClose={handleMenuClose}
+                    style={styles.menuContainer}
+                >
+                    <ScrollView keyboardShouldPersistTaps="always">
+                        {toggleOptions
+                            ? toggleOptions.map(option => (
+                                  <Menu.Item
+                                      key={option.title}
+                                      title={option.title}
+                                      active={option.active}
+                                      closeMenuOnClick={false}
+                                      iconName={option.iconName}
+                                      onPress={() => {
+                                          handleToggleOptionPress(option.title);
+                                          option.onPress();
+                                      }}
+                                  />
+                              ))
+                            : filteredOptions.map((option, index) => (
+                                  <Menu.Item
+                                      key={index}
+                                      title={option}
+                                      onPress={() => {
+                                          setInputValue(option);
+                                          onSelect(option);
+                                          setIsOptionsVisible(false);
+                                      }}
+                                  />
+                              ))}
+                    </ScrollView>
+
+                    {/* <FlatList
+                        data={filteredOptions}
+                        keyExtractor={item => item}
+                        renderItem={({ item }) => (
+                            <TouchableOpacity
+                                style={styles.option}
+                                onPress={() => {
+                                    setInputValue(item);
+                                    onSelect(item);
+                                    setIsOptionsVisible(false);
+                                }}
+                            >
+                                <Text style={styles.optionText}>{item}</Text>
+                            </TouchableOpacity>
+                        )}
+                    /> */}
```

Comment:
This component is getting quite long. Consider separating it into multiple components